packages/local-environment/src/network.ts

```diff
@@ -1,359 +1,596 @@
-import Docker from "dockerode";
-import { ethers } from "ethers";
-
-import { Key } from ".";
+import { Agents, AgentType } from "./agent";
+import { NetworkSpecs, ContractConfig, BridgeConfiguration, BaseAgentConfig, Domain, CoreContracts, BridgeContracts, AgentConfig, LogConfig, AppConfig, NomadGasConfig } from '@nomad-xyz/configuration';
+import { DockerizedActor } from "./actor";
+import Dockerode from "dockerode";
 import { sleep } from "./utils";
-import { DockerizedActor } from "./actors";
-import { SignerWithAddress } from "@nomiclabs/hardhat-ethers/signers";
-// TODO: the old deploy (what is currently published to npm) has been deprecated
-import { Governor } from "@nomad-xyz/deploy/src/core/CoreDeploy";
-
-export type Networkish = string | number | Network;
-
-enum NetworkStatus {
-  Running,
-  Stopped,
-  Disconnected,
-  Waiting, // Transition period
-}
-
-interface LocalNetworkHandler {
-  connect(): Promise<boolean>;
-  disconnect(): Promise<void>;
-  start(): Promise<void>;
-  stop(): Promise<void>;
-  status(): Promise<NetworkStatus>;
-  addKeys(...ks: Key[]): void;
-}
+import { ethers } from 'ethers';
+import { Key } from "./key";
+import { Env } from "./le";
+import {  } from "@nomad-xyz/configuration";
+//import { getContractAddress } from "ethers/lib/utils";
 
-class DockerNetworkHandler
-  extends DockerizedActor
-  implements LocalNetworkHandler
-{
-  network: Network;
-  keys: Key[];
-
-  constructor(network: Network) {
-    super(network.name, "net");
-    this.network = network;
-    this.keys = [];
+enum DockerNetworkStatus {
+    Running,
+    Stopped,
+    Disconnected,
+    Waiting, // Transition period
   }
 
-  addKeys(...ks: Key[]) {
-    this.keys.push(...ks);
-  }
-
-  async createContainer(): Promise<Docker.Container> {
-    const name = this.containerName();
-    const port = this.network.location.port;
-
-    return this.docker.createContainer({
-      Image: "hardhat",
-      name,
-      Env: [
-        "BLOCK_TIME=300",
-        ...this.keys.map((k, i) => `PRIVATE_KEY${i + 1}=${k.toString()}`),
-      ],
-      ExposedPorts: {
-        "8545/tcp:": {},
-      },
-      HostConfig: {
-        PortBindings: {
-          "8545/tcp": [
-            {
-              HostPort: port.toString(),
+export abstract class Network {
+    agents?: Agents;
+
+    connectedNetworks: Network[];
```

Comment:
Thinking we should move the concept of "connections" to the higher-level abstraction